DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/Startup.cs

```diff
@@ -46,6 +46,7 @@ public void Configuration(IAppBuilder app)
 
             ConfigureOauth(app);
             ConfigureCookies(app);
+            app.Use<TokenCheckMiddleware>();
```

Comment:
Yes, you are right. It's checking if user is authenticated. If he is it's checks for external access token existence if it is not present we are rejecting the request. With this middleware we can be sure that external access token is present in claims, so we don't need to check for it's existence in controller where it is used